### ISSUE TITLE
Broken since docker 26

### ISSUE THREAD

**Issue by sbp-bvanb** [wagoodman/dive#534]
**What happened**:

Github ubuntu 20.04 updated the docker version from 24 to 26

**What you expected to happen**:

dive continued to work

**How to reproduce it (as minimally and precisely as possible)**:

Run dive with docker 26

**Anything else we need to know?**:

**Environment**:
- OS version
- Docker version (if applicable)

**Comment by black-snow** [comment]
```
Fetching image... (this can take a while for large images)
cannot fetch image
could not find 'blobs/sha256/whatever' in parsed layers
```

For me this _seems_ to be related to containerd storage engine. I think it brokes as I switched to it.

**Comment by alastair87** [comment]
I agree re. containerd storage engine. I had containerd enabled on Docker Desktop with Docker v26.1.4 on Windows 11 with WSL2 - it was broken with containerd enabled but disabling containerd fixed it. It would be helpful if it was compatible with containerd though.

**Comment by calebAtIspot** [comment]
Anyone know how to download and run a earlier docker version on mac? Brew only has the latest version. I tried downloading from https://download.docker.com/mac/static/stable/aarch64/but got `"“docker” can’t be opened because Apple cannot check it for malicious software.` when I tried running it.

**Comment by calebAtIspot** [comment]
ah found the install instructions: https://docs.docker.com/engine/install/binaries/

In my case running `xattr -rc docker` fixed the error I was getting

**Comment by max-allan** [comment]
Disabling containerd in Docker Desktop settings on OSX has fixed it for me. (With latest version of Docker Desktop : Docker version 27.0.2, build 912c1ddf8a)
But it would be ideal to be compatible with containerd too.

**Comment by froody** [comment]
The issue is the heuristic [here](https://github.com/wagoodman/dive/blob/main/dive/image/docker/image_archive.go#L105) is broken because newer oci layers do gzip small layers, so they are skipped because they don't fill the buffer. Commenting out that check means I can load `gcr.io/distroless/nodejs20-debian12:latest` which I couldn't before.

**Comment by froody** [comment]
I hacked around it, pushed my changes here: https://github.com/froody/dive/tree/crane

quite a significant change, basically uses crane as a dest for docker to "push" to, much faster downloads, and now we have the manifest/config ahead of time to decide on the file format

**Comment by reneleonhardt** [comment]
> I hacked around it, pushed my changes here: https://github.com/froody/dive/tree/crane
> 
> quite a significant change, basically uses crane as a dest for docker to "push" to, much faster downloads, and now we have the manifest/config ahead of time to decide on the file format

Wow, thank you for your work! Can you make a PR?

**Comment by froody** [comment]
@reneleonhardt ok I'll try

**Comment by reneleonhardt** [comment]
Thank you, let me know if I can help 🙂

**Comment by bredamatt** [comment]
Disabling containerd does it on OSX (Docker Desktop).

**Comment by DeanAyalon** [comment]
> Disabling containerd does it on OSX (Docker Desktop).

Instead of disabling it, you can also use DinD:

```sh
docker run -d --name dind --privileged docker
docker exec -it dind sh
# within dind
docker run --rm -it -v /var/run/docker.sock:/var/run/docker.sock wagoodman/dive [image]    # or jauderho/dive for ARM64
```

**Comment by DeanAyalon** [comment]
Could not build froody's binary
Issue seems to be resolved with [Maddog2050's fork](https://github.com/Maddog2050/dive/tree/fix-oci-format)

**Comment by willcl-ark** [comment]
🎉
